code/modules/modular_computers/file_system/programs/messenger/messenger_stuff.dm

```diff
@@ -0,0 +1,148 @@
+/**
+ * I'll come up with a better file name later
+ */
+
+/// A list of all active and visible messengers
+GLOBAL_LIST_EMPTY_TYPED(TabletMessengers, /datum/computer_file/program/messenger)
+
+/// Registers an NTMessenger instance to the list of TabletMessengers.
+/proc/add_messenger(datum/computer_file/program/messenger/msgr)
+	var/obj/item/modular_computer/messenger_device = msgr.computer
+	// a bunch of empty PDAs are normally allocated, we don't want that clutter
+	if(!messenger_device.saved_identification || !messenger_device.saved_job)
+		return
+
+	if(!istype(msgr))
+		return
+
+	var/msgr_ref = REF(msgr)
+	if(msgr_ref in GLOB.TabletMessengers)
+		return
+
+	GLOB.TabletMessengers[msgr_ref] = msgr
+
+/// Unregisters an NTMessenger instance from the TabletMessengers table.
+/proc/remove_messenger(datum/computer_file/program/messenger/msgr)
```

Comment:
I don't see where this is being called.

Reply:
This is called by the messenger when the imprint on its PDA is reset, for unregistering from the table.